docs/source/developers/cpp/building.rst

```diff
@@ -624,13 +624,6 @@ outputs like:
    ../src/arrow/ipc/ipc-read-write-test.cc:574 code: writer->WriteRecordBatch(batch)
    NotImplemented: Unable to convert type: decimal(19, 4)
 
-Deprecations and API Changes
-----------------------------
-
-We use the compiler definition ``ARROW_NO_DEPRECATED_API`` to disable APIs that
-have been deprecated. It is a good practice to compile third party applications
-with this flag to proactively catch and account for API changes.
```

Comment:
Can we mention the `-Werror=deprecated-declarations` option instead of removing this section entirely?

Reply:
It seems we need to mention different kinds of compilers as the flag varies.

Reply:
https://github.com/apache/arrow/blob/299eb26e8c22b4aad4876c9e3b52f9adde699a5c/cpp/src/arrow/util/macros.h#L136

Is the definition of `__COVERITY__` still used? I cannot find it anywhere else. @kou

Reply:
@pitrou Are we still use Coverity?

Reply:
No. We've only had once access to a Coverity report of Arrow C++, a long time ago. It actually didn't surface any major issue.